src/boot/chid.c

```diff
@@ -112,9 +115,11 @@ EFI_STATUS chid_match(const void *hwid_buffer, size_t hwid_length, const Device
         /* Count devices and check validity */
         for (; (n_devices + 1) * sizeof(*devices) < hwid_length;) {
 
+                uint32_t dev_type;
+                dev_type = DEVICE_TYPE_FROM_DESCRIPTOR(devices[n_devices].descriptor);
                 if (devices[n_devices].descriptor == DEVICE_DESCRIPTOR_EOL)
                         break;
-                if (devices[n_devices].descriptor != DEVICE_DESCRIPTOR_DEVICETREE)
+                if ((dev_type != DEVICE_TYPE_FW) && (dev_type != DEVICE_TYPE_DEVICETREE))
```

Comment:
use IN_SET()

Reply:
This has already been fixed in the latest version.

Reply:
See https://github.com/systemd/systemd/pull/35747